aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -510,4 +534,54 @@ Tensor& capped_random_cuda_(Tensor& self, int64_t to, Generator* gen) {
   return clamped_random_cuda_(self, 0, to, gen);
 }
 
+Tensor& normal_cuda_(Tensor& self, double mean, double std, Generator* gen) {
+  TORCH_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  auto iter = TensorIterator::nullary_op(self);
+  normal_kernel_cuda(*iter, mean, std, gen);
+  return self;
+}
+
+Tensor& normal_out_cuda(Tensor& output, const Tensor& mean_, double std, Generator* gen) {
+  auto mean = std::get<0>(expand_inplace(output, mean_.to(kCUDA)));
```

Comment:
This looks wrong. We never do implicit conversions from CPU tensors to CUDA tensors. Is there a reason you added this line here?

Reply:
I don't see any indication this happened in the old code either

Reply:
I was just following how the bernoulli kernel was ported from THC to ATen. Don't really know if that's needed or not.
```
Tensor& bernoulli_tensor_cuda_(Tensor &self, const Tensor& p_, Generator* gen) {
  auto p = std::get<0>(expand_inplace(self, p_.to(kCUDA)));
```

Reply:
May be we were doing the expand like this? `THCTensor_(resizeAs)(state, self, means);	`

Reply:
Let me try removing them and see what happens.

Reply:
Ok looks like removing expand_inplace works fine. Curious to know why it was there in bernoulli though @SsnL

Reply:
Frankly speaking, I don't know. I copied that behavior when I moved bernoulli to aten. However, maybe it could be related to broadcasting with `torch.bernoulli`?

Reply:
That's right, bernoulli_tensor_cuda_kernel cannot handle unexpanded p, however, at::mul and at::add can.

Reply:
Removed expand_inplaces.